python/ngen_cal/src/ngen/cal/ngen.py

```diff
@@ -108,6 +109,9 @@ def __init__(self, **kwargs):
         #Let pydantic work its magic
         super().__init__(**kwargs)
         #now we work ours
+        # Register the default ngen output hook
+        self._plugin_manager.register(TrouteOutput(self.routing_output))
```

Comment:
Maybe im just weird, but I think of the plugins like http middleware or handlers. Even if its a little "gross" I would rather have a single place in the codebase where all the plugins are registered if that is possible.

Reply:
This has several nuanced considerations, with valid concerns on both sides of the topic.  For now, I propose leaving this in and seeing how more hook semantics play out and how we might want to manage globally vs scoped/controlled instances.